File: solutions/clojure/atbash-cipher/1/src/atbash_cipher.clj
(ns atbash-cipher
 (:require [clojure.string :as st]))



(defn atb_encode [c] 
  (let [ic (int c)]
    (cond (>= ic 97 ) (char (- 122 (- ic 97)))
        (>= ic 48) c     
      )
    )
)

 
(defn encode [input] ;; <- arglist goes here
  ;; your code goes here
  (->>
    (st/replace input #"\W+" "")
    st/lower-case
    (map atb_encode)
    (partition-all 5)
    (map #(apply str %))
    (st/join " ")
    )
 )

File: solutions/julia/atbash-cipher/1/atbash-cipher.jl
function atbash(st)
    res =""
     for lt in st
        if lt in 'a':'z' 
            c = Char(Int('z') - (Int(lt) - Int('a')))
         elseif lt in '0':'9'
            c = lt
        else
            c=""
        end
        res = string(res,c)
    end
    res
end  

group5(s) = 
begin
    res=""
    i = 0
    for lt in s
        i+=1
        res=string(res,lt)
        if (mod(i, 5)==0) && i!=length(s) 
            res=string(res," ") 
        end
    end
    res
end

function encode(input)
     raw = atbash(lowercase(input))
     group5(raw)
end

function decode(input)
    atbash(lowercase(input))
end

